jerry-ext/context/include/jerry-context.h

```diff
@@ -0,0 +1,43 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_CONTEXT_H
+#define JERRYX_CONTEXT_H
+
+#include "jerryscript.h"
+
+#ifndef JERRYX_CONTEXT_SLOTS
+#define JERRYX_CONTEXT_SLOTS 1
+#endif /* !JERRYX_CONTEXT_SLOTS */
+
+/**
+ * Function to pass to jerry_init_with_user_context() as its second parameter
+ */
+void *jerryx_context_init (void);
```

Comment:
The confusing part is `use_context` is not a context, it is a user specific data space inside of a context.

`multiple-context` is also inappropriate. We have PR #1778 about multiple-context, which means jerry can switch whole context to another. But here, all the slots are in one context.

I think `user_data` is better than `user_context`.

Any ideas? @martijnthe